docs/conceptual/core/executor.md

```diff
@@ -2,8 +2,91 @@
 title: Executor
 ---
 
-* Runtime Versioning
+The executor is responsible for dispatching and executing calls into the Substrate runtime.
 
-* Native VS Wasm Execution
+## Runtime Execution
 
-* Forkless Runtime Upgrades
\ No newline at end of file
+The Substrate runtime is compiled into a native executable and a WebAssembly (Wasm) binary.
+
+The native runtime is included as part of the node executable, while the Wasm binary is stored on the blockchain under a well known storage key.
+
+These two representations of the runtime may not be the same. For example: when the runtime is upgraded. The executor determines which version of the runtime to use when dispatching and executing calls.
+
+### Execution Strategy
```

Comment:
Conceptually, this happens above the executor. The executor just gets the information on how a function should be executed.

Reply:
Hmm, is this true?

We are talking about the Execution strategy here or the version lookup?

In the code, I see that `executor/src/native_executor.rs` calls the function `can_call_with`

Which ultimately does the version lookup logic:

```
	pub fn can_call_with(&self, other: &RuntimeVersion) -> bool {
		self.spec_version == other.spec_version &&
		self.spec_name == other.spec_name &&
		self.authoring_version == other.authoring_version
	}
```

This is what I meant by it "deciding"

Reply:
I think this topic is relevant, so I will simply specify that this happens from the client when it calls the executor. Is that okay?

Reply:
I added "Before runtime execution begins, the Substrate client passes to the executor an execution strategy."

Reply:
The executor does not know anything about execution strategies. Just if it should try to use native, not more.

Reply:
> Before runtime execution begins, the Substrate client chooses which runtime execution environment should be used. This is controlled by the execution strategy, which can be configured for the different parts of the blockchain execution process. The strategies are: ...

Better? Again, I would want to confirm that you agree talking about the execution strategy stuff totally makes sense here, even though it happens above the executor.

I just need to make sure the wording is accurate and understandable.

Reply:
Yeah that is okay. It is almost good. Could we just replace "choose" with "propose" or something similar. Because, even when you choose native, but the executor finds a version mismatch, it executes it in wasm.